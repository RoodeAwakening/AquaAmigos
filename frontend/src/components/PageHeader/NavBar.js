import React from 'react';
import { NavLink } from 'react-router-dom';
import LogoutButton from '../auth/LogoutButton';
import styles from './NavBar.module.css';

const NavBar = () => {
  return (
    <nav className={styles.NavBar_container}>
      <ul>
        <li>
          <NavLink to="/" exact activeClassName="active">
            Home
          </NavLink>
        </li>
        <li>
          <NavLink to="/login" exact activeClassName="active">
            Login
          </NavLink>
        </li>
        <li>
          <NavLink to="/sign-up" exact activeClassName="active">
            Sign Up
          </NavLink>
        </li>
        <li>
          <NavLink to="/users" exact activeClassName="active">
            Users
          </NavLink>
        </li>
        <li>
          <LogoutButton />
        </li>
      </ul>
    </nav>
  );
};

export default NavBar;
