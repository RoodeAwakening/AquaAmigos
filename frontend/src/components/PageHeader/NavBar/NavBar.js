import { NavLink } from 'react-router-dom';
import i18next from 'i18next';
import React from 'react';
import MembersOnly from '../../securityRoutes/MembersOnly/MembersOnly';
import styles from './NavBar.module.css';

const NavBar = () => {
  return (
    <nav className={styles.NavBar_container}>
      <ul>
        <li>
          <NavLink to="/" exact activeClassName="active">
            <button type="button" className={styles.btn}>
              {i18next.t('msg_home')}
            </button>
          </NavLink>
        </li>
        <MembersOnly>
          <li>
            <NavLink to="/topics" exact activeClassName="active">
              <button type="button" className={styles.btn}>
                {i18next.t('msg_message_board')}
              </button>
            </NavLink>
          </li>
        </MembersOnly>
        <li>
          <NavLink to="/media" exact activeClassName="active">
            <button type="button" className={styles.btn}>
              {i18next.t('msg_media')}
            </button>
          </NavLink>
        </li>
        <li>
          <NavLink to="/events" exact activeClassName="active">
            <button type="button" className={styles.btn}>
              {i18next.t('msg_events')}
            </button>
          </NavLink>
        </li>
        <li>
          <NavLink to="/news" exact activeClassName="active">
            <button type="button" className={styles.btn}>
              {i18next.t('msg_news')}
            </button>
          </NavLink>
        </li>
        <li>
          <NavLink to="/about-us" exact activeClassName="active">
            <button type="button" className={styles.btn}>
              {i18next.t('msg_about')}
            </button>
          </NavLink>
        </li>
        <MembersOnly>
          <li>
            <NavLink to="/users" exact activeClassName="active">
              <button type="button" className={styles.btn}>
                {i18next.t('msg_members')}
              </button>
            </NavLink>
          </li>
        </MembersOnly>
        <MembersOnly>
          <li>
            <NavLink to="/email" exact activeClassName="active">
              <button type="button" className={styles.btn}>
                {i18next.t('msg_email')}
              </button>
            </NavLink>
          </li>
        </MembersOnly>

      </ul>
    </nav>
  );
};

export default NavBar;
