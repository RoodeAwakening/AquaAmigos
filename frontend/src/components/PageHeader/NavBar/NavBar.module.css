.NavBar_container {
  display: flex;
  justify-content: space-evenly;
}

.NavBar_container ul {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding: 0rem 1rem;
  background-color: #fff;
  border-bottom: 1px solid #e5e5e5;
  margin: .5rem 0rem;
}

.NavBar_container ul li {
  list-style: none;
  margin: .5rem .5rem;

}

.btn {
  text-transform: uppercase;
  border-radius: 10rem;
  text-decoration: none;
  padding: 0.5rem 2rem;
  display: inline-block;
  transition: all 0.2;
  position: relative;
  font-size: 0.6rem;
}
.btn:link,
.btn .bnt:visited {
  text-transform: uppercase;
  text-decoration: none;
  padding: 1.5rem 4rem;
  display: inline-block;
  border-radius: 10rem;
  transition: all 0.2;
  position: relative;
  font-size: 1.6rem;
}
.btn:hover {
  transform: translateY(-0.3rem);
  box-shadow: 0 1rem 2rem rgba(0, 0, 0, 0.226);
}
.btn:hover::after {
  transform: scaleX(1.4) scaleY(1.6);
  opacity: 0;
}
.btn:active {
  transform: translateY(-0.1rem);
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.226);
}
.btn--white {
  background-color: #fff f;
  color: #4e4949;
}
.btn--white::after {
  background-color: #fff f;
}
.btn::after {
  content: "";
  display: inline-block;
  height: 100%;
  width: 100%;
  border-radius: 10rem;
  position: absolute;
  top: 0;
  left: 0;
  z-index: -1;
  transition: all 0.4s;
}
.btn--animated {
  animation: moveInButton 0.5s ease-out 0.75s;
  /* applies the 0% to the object */
  animation-fill-mode: backwards;
}


@media screen and (max-width: 768px){
  .NavBar_container ul{
    display: flex;
    flex-direction: column;
  }
}