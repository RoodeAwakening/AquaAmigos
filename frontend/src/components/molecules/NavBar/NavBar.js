import { NavLink } from 'react-router-dom';
import i18next from 'i18next';
import React from 'react';
import PropTypes from 'prop-types';
import MembersOnly from '../../securityRoutes/MembersOnly/MembersOnly';
import navLinks from './links';
import styles from './NavBar.module.css';
import NavButton from '../../atoms/NavButton/NavButton';

const NavBar = ({ showMenu }) => {
//  a list of link routes and titles to map through
// if you need to add another link please add it to the links.js file
  const links = navLinks.map((link) => {
    return (
      link.memberOnly ? (
        <MembersOnly>
          <li>
            <NavLink to={link.link} exact activeClassName="active">
              <NavButton onClick={showMenu}>
                {i18next.t(link.title)}
              </NavButton>
            </NavLink>
          </li>
        </MembersOnly>
      ) : (
        <li>
          <NavLink to={link.link} exact activeClassName="active">
            <NavButton onClick={showMenu}>
              {i18next.t(link.title)}
            </NavButton>
          </NavLink>
        </li>
      )
    );
  });

  return (
    <nav className={styles.NavBar_container}>
      <ul>
        {links}
      </ul>
    </nav>
  );
};

NavBar.defaultProps = {
  showMenu: {},
};

NavBar.propTypes = {
  showMenu: PropTypes.func,
};

export default NavBar;
