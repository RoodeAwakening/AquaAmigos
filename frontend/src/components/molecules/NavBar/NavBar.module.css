.NavBar_container {
  display: flex;
  justify-content: space-evenly;
}

.NavBar_container ul {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding: 0rem 1rem;
  background-color: #fff;
  border-bottom: 1px solid #e5e5e5;
  margin: .5rem 0rem;
}

.NavBar_container ul li {
  list-style: none;
  margin: .5rem .5rem;

}




@media screen and (max-width: 768px){
  .NavBar_container ul{
    display: flex;
    flex-direction: column;
  }
}