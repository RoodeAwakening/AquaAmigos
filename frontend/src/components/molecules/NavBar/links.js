export default [
  {
    link: '/',
    title: 'msg_home',
    memberOnly: false,
  },
  {
    link: 'topics',
    title: 'msg_message_board',
    memberOnly: true,
  },
  {
    link: '/media',
    title: 'msg_media',
    memberOnly: false,
  },
  {
    link: '/events',
    title: 'msg_events',
    memberOnly: false,
  },
  {
    link: '/news',
    title: 'msg_news',
    memberOnly: false,
  },
  {
    link: '/about-us',
    title: 'msg_about',
    memberOnly: false,
  },
  {
    link: '/users',
    title: 'msg_members',
    memberOnly: true,
  },
  {
    link: '/email',
    title: 'msg_email',
    memberOnly: true,
  },
];
