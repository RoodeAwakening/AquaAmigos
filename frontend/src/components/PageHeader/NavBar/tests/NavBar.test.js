import React from 'react';
import { shallow, configure } from 'enzyme';

import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

import NavBar from '../NavBar';

configure({ adapter: new Adapter() });
global.confirm = () => true;

function setup() {
  return shallow(<NavBar />);
}

describe('Navbar Snapshot', () => {
  let navbar;

  beforeEach(() => {
    navbar = setup();
  });

  afterEach(() => {
    navbar.unmount();
  });

  it('Should render the navbar to match the snapshot', () => {
    expect(toJson(navbar)).toMatchSnapshot();
  });
});
