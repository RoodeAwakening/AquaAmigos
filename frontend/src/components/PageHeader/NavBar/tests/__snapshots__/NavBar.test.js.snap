// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Snapshot Should render the navbar to match the snapshot 1`] = `
<nav
  className="NavBar_container"
>
  <ul>
    <li>
      <NavLink
        activeClassName="active"
        exact={true}
        to="/"
      >
        <button
          className="btn"
          type="button"
        />
      </NavLink>
    </li>
    <MembersOnly>
      <li>
        <NavLink
          activeClassName="active"
          exact={true}
          to="/topics"
        >
          <button
            className="btn"
            type="button"
          />
        </NavLink>
      </li>
    </MembersOnly>
    <li>
      <NavLink
        activeClassName="active"
        exact={true}
        to="/media"
      >
        <button
          className="btn"
          type="button"
        />
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="active"
        exact={true}
        to="/events"
      >
        <button
          className="btn"
          type="button"
        />
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="active"
        exact={true}
        to="/about-us"
      >
        <button
          className="btn"
          type="button"
        />
      </NavLink>
    </li>
    <MembersOnly>
      <li>
        <NavLink
          activeClassName="active"
          exact={true}
          to="/users"
        >
          <button
            className="btn"
            type="button"
          />
        </NavLink>
      </li>
    </MembersOnly>
    <MembersOnly>
      <li>
        <NavLink
          activeClassName="active"
          exact={true}
          to="/email"
        >
          <button
            className="btn"
            type="button"
          />
        </NavLink>
      </li>
    </MembersOnly>
  </ul>
</nav>
`;
